libs/tracing/Tracing.mli

```diff
@@ -2,10 +2,41 @@
    See header of Tracing.ml for details
  *)
 
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+type analysis_flags = {
+  secrets_validators : bool;
+  historical_scan : bool;
+  allow_all_origins : bool;
+  deep_intra_file : bool;
+  deep_inter_file : bool;
+}
+
+type top_level_data = { version : string; analysis_flags : analysis_flags }
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+val oss_analysis : unit -> analysis_flags
+(** For analysis run with the oss engine, we know all the flags will be false *)
```

Comment:
Based on description I might rather call this `no_analysis_features` or similar.